pkg/operator/controller/controller_test.go

```diff
@@ -216,69 +216,9 @@ var _ = Describe("Controller", func() {
 			})
 
 			It("should become ready", func() {
-				one := int32(1)
 				args := createArgs()
 				doReconcile(args)
-
-				resources, err := getAllResources(args.reconciler)
-				Expect(err).ToNot(HaveOccurred())
-
-				for _, r := range resources {
-					d, ok := r.(*appsv1.Deployment)
-					if !ok {
-						continue
-					}
-
-					numReplicas := d.Spec.Replicas
-					Expect(numReplicas).ToNot(BeNil())
-					Expect(numReplicas).To(Equal(&one))
-
-					d, err := getDeployment(args.client, d)
-					Expect(err).ToNot(HaveOccurred())
-					d.Status.Replicas = *numReplicas
-					err = args.client.Update(context.TODO(), d)
-					Expect(err).ToNot(HaveOccurred())
-
-					doReconcile(args)
-
-					Expect(args.cdi.Status.Conditions).Should(HaveLen(3))
-					Expect(conditions.IsStatusConditionTrue(args.cdi.Status.Conditions, conditions.ConditionAvailable)).To(BeTrue())
-					Expect(conditions.IsStatusConditionFalse(args.cdi.Status.Conditions, conditions.ConditionProgressing)).To(BeTrue())
-					// We will expect degraded status, because in the test the deployment.status.replicas will not be 1, when the desired is 1.
-					Expect(conditions.IsStatusConditionTrue(args.cdi.Status.Conditions, conditions.ConditionDegraded)).To(BeTrue())
-				}
-
-				resources, err = getAllResources(args.reconciler)
-				Expect(err).ToNot(HaveOccurred())
-				running := false
-
-				createUploadProxyCACertSecret(args.client)
-
-				for _, r := range resources {
-					d, ok := r.(*appsv1.Deployment)
-					if !ok {
-						continue
-					}
-
-					Expect(running).To(BeFalse())
-
-					d, err := getDeployment(args.client, d)
-					Expect(err).ToNot(HaveOccurred())
-					d.Status.ReadyReplicas = d.Status.Replicas
-					err = args.client.Update(context.TODO(), d)
-					Expect(err).ToNot(HaveOccurred())
-
-					doReconcile(args)
-
-					if len(args.cdi.Status.Conditions) == 3 &&
-						conditions.IsStatusConditionTrue(args.cdi.Status.Conditions, conditions.ConditionAvailable) &&
-						conditions.IsStatusConditionFalse(args.cdi.Status.Conditions, conditions.ConditionProgressing) &&
-						conditions.IsStatusConditionFalse(args.cdi.Status.Conditions, conditions.ConditionDegraded) {
-						running = true
-					}
-				}
-
-				Expect(running).To(BeTrue())
```

Comment:
lot of red here.  not sure how this is all useless/invalid now.

Reply:
That is a duplicate of setDeploymentsReady(args), the code is identical. So instead of duplicating it, I just replaced it.

Reply:
part of it is identical.  not the whole thing.

Reply:
Actually you are right, first it setting the Status.Replicas to a valid number, and then it checks what the conditions are, then it sets the ReadyReplicas and checks the conditions again. Where setDeploymentsReady sets both at once and then checks the conditions. If we are interested in the intermediate check, I can put it back.